pavelib/quality.py

```diff
@@ -301,47 +304,140 @@ def run_jshint(options):
+                count=safelint_counts['total'], violations_limit=violation_thresholds['total']
+            )
+
+    # Test rule violations against thresholds.
+    if 'rules' in violation_thresholds:
+        threshold_keys = sorted(violation_thresholds['rules'].keys())
+        for threshold_key in threshold_keys:
+            if threshold_key not in safelint_counts['rules']:
+                error_message += (
+                    "\nNumber of {safelint_script} violations for {rule} could not be found in "
+                    "{safelint_report}."
+                ).format(
+                    safelint_script=safelint_script, rule=threshold_key, safelint_report=safelint_report
+                )
+            elif violation_thresholds['rules'][threshold_key] < safelint_counts['rules'][threshold_key]:
+                error_message += \
+                    "\nToo many {rule} violations ({count}).\nThe {rule} limit is {violations_limit}.".format(
+                        rule=threshold_key, count=safelint_counts['rules'][threshold_key],
+                        violations_limit=violation_thresholds['rules'][threshold_key],
+                    )
+
+    if error_message is not "":
+        raise BuildFailure(
+            "SafeTemplateLinter Failed.\n{error_message}\n"
+            "See {safelint_report} or run the following command to hone in on the problem:\n"
+            "  ./scripts/safe-commit-linter.sh -h".format(
+                error_message=error_message, safelint_report=safelint_report
+            )
+        )
+
+
+@task
+@needs('pavelib.prereqs.install_python_prereqs')
+def run_safecommit():
+    """
+    Runs safe-commit-linter.sh on the current branch.
+    """
+    safecommit_script = "safe-commit-linter.sh"
+    safecommit_report_dir = (Env.REPORT_DIR / "safecommit")
+    safecommit_report = safecommit_report_dir / "safecommit.report"
+    _prepare_report_dir(safecommit_report_dir)
+
+    sh(
+        "{repo_root}/scripts/{safecommit_script} >> {safecommit_report}".format(
+            repo_root=Env.REPO_ROOT,
+            safecommit_script=safecommit_script,
+            safecommit_report=safecommit_report,
+        ),
+        ignore_error=True
+    )
+
+    safecommit_count = _get_safecommit_count(safecommit_report)
+
+    try:
+        num_violations = int(safecommit_count)
+    except TypeError:
+        raise BuildFailure(
+            "Error. Number of {safecommit_script} violations could not be found in {safecommit_report}".format(
+                safecommit_script=safecommit_script, safecommit_report=safecommit_report
             )
         )
 
+    # Print number of violations to log.
+    violations_count_str = "Number of {safecommit_script} violations: {num_violations}".format(
+        safecommit_script=safecommit_script, num_violations=num_violations
+    )
+    print violations_count_str
+
+    # Record the metric
+    _write_metric(violations_count_str, (Env.METRICS_DIR / "safecommit"))
```

Comment:
@benpatterson Let's discuss the possibilities for this for a minute in person.  Thanks!